File: my-folder/0274-h-index/solution.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        counts = [0] * (n+1)

        for i in citations:
            if i > n:
                counts[n] += 1
            else:
                counts[i] += 1
        
        papers = 0

        for i in range(n, -1, -1):
            papers += counts[i]
            if papers >= i:
                return i
        
        return 0

File: my-folder/0274-h-index/solution.java
class Solution {
    public int hIndex(int[] citations) {
        int n = citations.length;
        int[] bucket = new int[n+1];
        for (int i : citations) {
            if (i < n) {
                bucket[i]++;
            }
            else {
                bucket[n]++;
            }
        }
        int count = 0;
        for (int i = n; i >= 0; i--) {
            count += bucket[i];
            if (count >= i) return i;
        }
        return -1;
    }
}
